ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.component.ts

```diff
@@ -162,35 +162,40 @@ export class ComputeCapacityComponent implements OnInit {
             Validators.max(memory['maxUsage'])
           ]);
 
-          // set max limit validator for endpointMemory
-          this.form.get('endpointMemory').setValidators([
-            ...getNumericValidatorsArray(false),
-            Validators.max(memory['maxUsage'])
-          ]);
+          if (this.inAdvancedMode) {
+            // set max limit validator for endpointMemory
+            this.form.get('endpointMemory').setValidators([
+              ...getNumericValidatorsArray(false),
+              Validators.max(memory['maxUsage'])
+            ]);
 
-          // set max limit validator for cpu unreservedForPool
-          this.form.get('cpuReservation').setValidators([
-            ...getNumericValidatorsArray(false),
-            Validators.max(cpu['unreservedForPool'])
-          ]);
+            // set max limit validator for cpu unreservedForPool
+            this.form.get('cpuReservation').setValidators([
+              ...getNumericValidatorsArray(false),
+              Validators.max(cpu['unreservedForPool'])
+            ]);
 
-          // set max limit validator for memory unreservedForPool
-          this.form.get('memoryReservation').setValidators([
-            ...getNumericValidatorsArray(false),
-            Validators.max(memory['unreservedForPool'])
-          ]);
+            // set max limit validator for memory unreservedForPool
+            this.form.get('memoryReservation').setValidators([
+              ...getNumericValidatorsArray(false),
+              Validators.max(memory['unreservedForPool'])
+            ]);
+
+            // This prevents the next button from getting disabled when the user selects a host or cluster that has less RAM
+            // available for VM endpoint than the default value. As a solution, we set the smaller value between the default
+            // value and memory['maxUsage']
+            this.form.get('endpointMemory').setValue(Math.min(memory['maxUsage'], endpointMemoryDefaultValue) + '');
+          } else {
+            this.form.get('endpointMemory').setValidators([]);
+            this.form.get('cpuReservation').setValidators([]);
+            this.form.get('memoryReservation').setValidators([]);
+          }
 
           this.form.get('cpuLimit').updateValueAndValidity();
           this.form.get('memoryLimit').updateValueAndValidity();
           this.form.get('endpointMemory').updateValueAndValidity();
           this.form.get('cpuReservation').updateValueAndValidity();
           this.form.get('memoryReservation').updateValueAndValidity();
```

Comment:
Are we updating value and validity for all fields on this form? if so maybe use something like this.
for (var i in this.form.controls) {
  this.form.get[i].updateValueAndValidity();
}

Reply:
No, we aren't. Only for those we change its validators when advanced settings are activated. I don't know why we doing it this way here, in others steps we are deactivating fields in which we don't want to consider their validations. Maybe I can do the same here, and group all these fields in a formGroup and deactivate that group.